src/infer_request.cc

```diff
@@ -1169,6 +1170,79 @@ InferenceRequest::Normalize()
         input.MutableShapeWithBatchDim()->push_back(d);
       }
     }
+    // Matching incoming request's shape and byte size to make sure the
+    // payload contains correct number of elements.
+    // Note: Since we're using normalized input.ShapeWithBatchDim() here,
+    // make sure that all the normalization is before the check.
+    {
+      const size_t& byte_size = input.Data()->TotalByteSize();
+      const auto& data_type = input.DType();
+      const auto& input_dims = input.ShapeWithBatchDim();
+      int64_t expected_byte_size = INT_MAX;
+      // Because Triton expects STRING type to be in special format
+      // (prepend 4 bytes to specify string length), so need to add all the
+      // first 4 bytes for each element to find expected byte size
+      if (data_type == inference::DataType::TYPE_STRING) {
+        int64_t element_count = triton::common::GetElementCount(input_dims);
+        int64_t element_idx = 0;
+        expected_byte_size = 0;
+        for (size_t i = 0; i < input.Data()->BufferCount(); ++i) {
+          BufferAttributes* buffer_attributes;
+          const char* content = input.Data()->BufferAt(i, &buffer_attributes);
+          size_t content_byte_size = input.Data()->TotalByteSize();
+
+          while (content_byte_size >= sizeof(uint32_t)) {
```

Comment:
I don't think this logic scales well for the case where `BufferCount() > 1`. The `content_byte_size` would return the bytesize for the complete input tensor including all the buffers.

Reply:
Fixed